### PR TITLE
[BACKLOG-33354] Update XML Output step to use enhanced VFS browser fo…

### PR DESCRIPTION
**Pull request by angelorod**
…r browsing and selecting files

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/xml/core
```
---
:ok_hand: All tests passed!
 
Tests run: **75**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-7269/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 29s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/xml/core
```
---
:ok_hand: All tests passed!
 
Tests run: **75**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-7269/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/xml/core/src/main/java/org/pentaho/di/ui/trans/steps/xmloutput/XMLOutputDialog.java [hunk 1]
```diff
@@ -848,22 +851,17 @@ public void modifyText( ModifyEvent e ) {
...
+    wbFilename.addSelectionListener(
+            new SelectionAdapterFileDialogTextVar(
+                    log,
+                    wFilename,
+                    transMeta,
+                    new SelectionAdapterOptions(
+                            SelectionOperation.SAVE_TO,
+                            new FilterType[] { FilterType.XML, FilterType.TXT, FilterType.CSV, FilterType.ALL },
+                            FilterType.XML,
+                            new String[] {"local"},
```
Comment (cseverino789) [inline plugins/xml/core/src/main/java/org/pentaho/di/ui/trans/steps/xmloutput/XMLOutputDialog.java]: There's a constructor that will take ProviderFilterType Enum, check line 63 of SelectionAdapterOptions

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 31s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/xml/core
```
---
:ok_hand: All tests passed!
 
Tests run: **75**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-7269/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 40s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/xml/core
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/xml/core
```
---
:ok_hand: All tests passed!
 
Tests run: **75**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/pdi/job/PR-7269/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**